apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/SamhsaMatcherR4FromClaimTransformerV2Test.java

```diff
@@ -0,0 +1,845 @@
+    if (claim instanceof PartDEvent
+        || claim instanceof DMEClaim
+        || claim instanceof HHAClaim
+        || claim instanceof HospiceClaim
+        || claim instanceof CarrierClaim) {
+      expectMatch = false;
+    }
+
+    verifySamhsaMatcherForProcedureIcd(
+        IcdCode.CODING_SYSTEM_ICD_9,
+        BLACKLISTED_IC9_PROCEDURE_CODE,
+        expectMatch,
+        loadedExplanationOfBenefit);
+  }
+
+  /**
+   * Verifies that when transforming a SAMHSA claim into an ExplanationOfBenefit (where the
+   * ExplanationOfBenefit then contains a procedure[n].coding[n].system =
+   * IcdCode.CODING_SYSTEM_ICD_9) and the ICD code is blacklisted the SAMHSA matcher's test method
+   * will identify this as a SAMHSA related ExplanationOfBenefit.
+   */
+  @Test
+  public void
+      testR4SamhsaMatcherWhenTransformedClaimHasProcedureWithUnknownIcd9SystemExpectMatch() {
+    boolean expectMatch = true;
+
+    // PDE has no SAMHSA, so expect no match on SAMSHA filter
+    // DME, HHA, Hospice, Carrier does not look at procedure so it wont match
+    if (claim instanceof PartDEvent
+        || claim instanceof DMEClaim
+        || claim instanceof HHAClaim
+        || claim instanceof HospiceClaim
+        || claim instanceof CarrierClaim) {
+      expectMatch = false;
+    }
+
+    verifySamhsaMatcherForProcedureIcd(
+        "not valid icd9 system", NON_BLACKLISTED_IC_CODE, expectMatch, loadedExplanationOfBenefit);
+  }
+
+  /**
+   * Verifies that when transforming a SAMHSA claim into an ExplanationOfBenefit (where the
+   * ExplanationOfBenefit then contains a procedure[n].coding[n].system =
+   * IcdCode.CODING_SYSTEM_ICD_9) and the ICD code is not blacklisted the SAMHSA matcher's test
+   * method will not identify this as a SAMHSA related ExplanationOfBenefit.
+   */
+  @Test
+  public void
+      testR4SamhsaMatcherWhenTransformedClaimHasProcedureWithNonBlacklistedIcd9CodeExpectNoMatch() {
+    verifySamhsaMatcherForProcedureIcd(
+        IcdCode.CODING_SYSTEM_ICD_9, NON_BLACKLISTED_IC_CODE, false, loadedExplanationOfBenefit);
+  }
+
+  /**
+   * Verifies that when transforming a SAMHSA claim into an ExplanationOfBenefit (where the
+   * ExplanationOfBenefit then contains a procedure[n].coding[n].system =
+   * IcdCode.CODING_SYSTEM_ICD_10) and the ICD code is blacklisted the SAMHSA matcher's test method
+   * will identify this as a SAMHSA related ExplanationOfBenefit.
+   */
+  @Test
+  public void
+      testR4SamhsaMatcherWhenTransformedClaimHasProcedureWithBlacklistedIcd10CodeExpectMatch() {
+    boolean expectMatch = true;
+
+    // PDE has no SAMHSA, so expect no match on SAMSHA filter
+    // DME, HHA, Hospice, Carrier does not look at procedure so it wont match
+    if (claim instanceof PartDEvent
+        || claim instanceof DMEClaim
+        || claim instanceof HHAClaim
+        || claim instanceof HospiceClaim
+        || claim instanceof CarrierClaim) {
+      expectMatch = false;
+    }
```

Comment:
Claim is actually dual-purpose:
1. Determine which claimtype we're using, which to your point could be done with the type field and the util
2. Have a param list with which to tie the testname to, see above comment

If I didn't need the test name as well I'd be inclined to nix it and use the type field, but since it also names the tests and we have no better option to fix the display name that I know of, I think we should keep it as-is for now.

Reply:
See previous comment on test case names.

Reply:
Changed to be a string parameter, which cleans up the test names.